src/openrct2-ui/windows/Ride.cpp

```diff
@@ -3651,6 +3665,44 @@ static void window_ride_operating_update(rct_window* w)
     }
 }
 
+static void window_ride_operating_set_min_waiting_time(rct_window* w, uint8_t time)
+{
+    auto rideSetSetting = RideSetSettingAction(w->number, RideSetSetting::MinWaitingTime, time);
+    GameActions::Execute(&rideSetSetting);
+}
+
+static void window_ride_operating_set_max_waiting_time(rct_window* w, uint8_t time)
+{
+    auto rideSetSetting = RideSetSettingAction(w->number, RideSetSetting::MaxWaitingTime, time);
+    GameActions::Execute(&rideSetSetting);
+}
+
+static void window_ride_operating_textinput(rct_window* w, rct_widgetindex widgetIndex, char* text)
+{
+    if ((widgetIndex != WIDX_MINIMUM_LENGTH && widgetIndex != WIDX_MAXIMUM_LENGTH) || text == nullptr)
+        return;
+
+    char* end;
+    uint8_t time = strtol(text, &end, 10);
+
+    if (*end == '\0')
+    {
+        uint8_t minTime = 0;
+        uint8_t maxTime = 250;
+        time = std::clamp(time, minTime, maxTime);
+
+        // if widget is min waiting time set it, else set max waiting time
+        if (widgetIndex == WIDX_MINIMUM_LENGTH)
```

Comment:
I prefer a switch case here. After all, we might want to add inputs for other widgets on that tab later on.